flask/app/genes.py

```diff
@@ -0,0 +1,104 @@
+from csv import DictWriter, QUOTE_MINIMAL
+from io import StringIO
+import traceback
+from flask import abort, jsonify, request, Blueprint, Response, send_file
+from flask_login import login_required
+from sqlalchemy import func
+from .extensions import db
+from .models import Gene
+from .utils import paged
+
+genes_blueprint = Blueprint(
+    "genes",
+    __name__,
+)
+
+
+@genes_blueprint.route("/api/summary/genes", methods=["GET"])
+@paged
+@login_required
+def genes(page, limit):
+    """ index route for genes """
+    search = request.args.get("search", type=str)
+    gene_query = db.session.query(Gene)
+    if search:
+        gene_query = gene_query.filter(func.instr(Gene.hgnc_gene_name, search))
+
+    total_count = gene_query.with_entities(
+        func.count(func.distinct(Gene.gene_id))
+    ).scalar()
+    if limit and page:
+        gene_query.limit(limit).offset(page * limit)
+    results = gene_query.all()
```

Comment:
if a gene does not have a hit in the database the response is : 
```
{
  "data": [], 
  "page": 0, 
  "total_count": 0
}
```
should there be a check for whether a result is returned from the database? eg. if len(results) == 0 then abort.

Reply:
I don't think so. Generally I think that the reasoning is that the client is asking the server for the results of a search against an index route, and the server is (in this case at least) obliged to send back a json representation of the results. In the case of no matches, the results are an empty json array. A 400 implies that the client has made an error, but I wouldn't construe an empty search as an error -- maybe an empty result set is informative or even exactly what they were looking for! A 400 (or 404) might be confusing in this case, making the client think they sent a bad param or tried a non-existent endpoint. This is different than, say, asking for a specific value, like `[GET] api/foo/2` where `2` is the id of some model `foo`. If the user asks for a model that doesn't exist, a 404 might be a good response (some frameworks return 404 automatically here). If there's persuasive documentation suggesting an error is is appropriate for an empty result set on an index route search, I can certainly be persuaded, though I haven't seen this implementation much in the wild :slightly_smiling_face: .

Reply:
i appreciate the detailed answer - that makes sense to me!